plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -246,32 +248,75 @@ public function menu_highlight() {
 				}
 				break;
 		}
+
+		// Highlight "Add new order" submenu when on the new order page.
+		if ( isset( $_GET['page'], $_GET['action'] ) && 0 === strpos( sanitize_text_field( wp_unslash( $_GET['page'] ) ), 'wc-orders' ) && 'new' === sanitize_text_field( wp_unslash( $_GET['action'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$submenu_file = add_query_arg( 'action', 'new', admin_url( 'admin.php?page=' . sanitize_text_field( wp_unslash( $_GET['page'] ) ) ) ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited,WordPress.Security.NonceVerification.Recommended
+		}
+	}
+
+	/**
+	 * Generate order count badge HTML.
+	 *
+	 * @param int $order_count The number of processing orders.
+	 * @return string The badge HTML.
+	 */
+	private function get_order_count_badge( $order_count ) {
+		return ' <span class="awaiting-mod update-plugins count-' . esc_attr( $order_count ) . '"><span class="processing-count">' . number_format_i18n( $order_count ) . '</span></span>';
 	}
 
 	/**
 	 * Adds the order processing count to the menu.
 	 */
 	public function menu_order_count() {
-		global $submenu;
+		global $submenu, $menu;
 
+		// Always remove 'WooCommerce' sub menu item.
 		if ( isset( $submenu['woocommerce'] ) ) {
-			// Remove 'WooCommerce' sub menu item.
 			unset( $submenu['woocommerce'][0] );
+		}
 
-			// Add count if user has access.
-			if ( apply_filters( 'woocommerce_include_processing_order_count_in_menu', true ) && current_user_can( 'edit_others_shop_orders' ) ) {
-				$order_count = apply_filters( 'woocommerce_menu_order_count', wc_processing_order_count() );
+		/**
+		 * Filters whether to include the processing order count in the menu.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param bool $include_count Whether to include the count. Default true.
+		 */
+		if ( ! apply_filters( 'woocommerce_include_processing_order_count_in_menu', true ) || ! current_user_can( 'edit_others_shop_orders' ) ) {
+			return;
+		}
```

Comment:
Did a double-take with this condition change but seems okay, it's just doing an early return instead of wrapping everything in a condition like the previous iteration.